File: Card Project 3/card.py
"""
This module builds a card, and provides information on the cards.

@author: Kristina Striegnitz, Roderick Landreth

Created September 26th 2018 for Card Project 2
Edited October 9th 2018 for Card Project 3

I affirm that I have carried out my
academic endeavors with full academic honesty. Roderick Landreth
"""


class Card:
    """Create a card, able to return only suit or only value"""
    suit_names = {"S":"Spades", "C": "Clubs","D": "Diamonds","H": "Hearts"}
    value_names = {1: 1,
                   2: 2,
                   3: 3,
                   4: 4,
                   5: 5,
                   6: 6,
                   7: 7,
                   8: 8,
                   9: 9,
                   10: 10,
                   11: "Jack",
                   12: "Queen",
                   13: "King",
                   14: "Ace"}

    def __init__(self,  suit, value):
        self.value = value
        self.suit = suit

    def __str__(self):
        """Return a string of how you would read the card."""
        return "{:5^} of {:5^}      ".format(self.value_names[int(self.value)], self.suit_names[self.suit])

    def card(self):
        """Return a fast way to look at a card's data."""
        return self.suit, self.value

File: Java/Card Project 4/Card.java
/**This class creates cards and dispenses information about a card object.
 *
 * @author Roderick Landreth
 * Created: November 9, 2018
 *
 I affirm that I have carried out my
 academic endeavors with full academic honesty. Roderick Landreth
 """
*/
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Dictionary;


public class Card {
    private ArrayList<String> VALUES = new ArrayList<String>(
            Arrays.asList("2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace") );
    //private Dictionary SUITS.put('H', "Poop");
    private char Suits;
    private int Value;

    public Card(char Suits, int Value){
        this.Suits = Suits;
        this.Value = Value;
    }

    /**Return the value of a given card.
     *
     * @return card's value
     */
    public int getRank(){
        return Value;
    }

    /**Return the suit of a given card.
     *
     * @return card's suit
     */
    public char getSuit(){
        return Suits;
    }

    /**Provide a string in the format of a card's name.
     *
     * @return a formatted string of the card representation.
     */
    public String toString(){
    return VALUES.get(Value-2)+" of " + Suits;
    }

    /**Create a copy of the card and return it, so when a card is returned the deck or hand cannot be altered
     * outside the program. Both instance variables are primitives, so a copy of their value in memory is returned.
     *
     * @return a copy of the card in question, for information hiding purposes.
     */
    public Card copy(){
        return new Card(Suits, Value);
    }

    /*public static void main(){
        Card newCard = new Card('S',3);
        System.out.println(newCard.getSuit());
        System.out.println(newCard.getRank());
        System.out.println(newCard.toString());
    }*/
}
